mltsp/util.py

```diff
@@ -81,3 +82,44 @@ def is_running_in_docker():
     else:
         in_docker_container = False
     return in_docker_container
+
+
+def cast_model_params(model_type, model_params):
```

Comment:
I can't quite follow the logic of this function.  Can you explain everything that it needs to do?